### PR TITLE
add annotations in pylint/checkers.

### PR DESCRIPTION
**Pull request by sushobhit27**
add mypy in tox.ini for only pylint/checkers.

### Fixes / new features
- #2079

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -17,12 +17,13 @@ deps =
...
    pytest
    pytest-xdist
-   pyenchant
+   mypy
 
 setenv =
     COVERAGE_FILE = {toxinidir}/.coverage.{envname}
 
 commands =
+    python -Wi {envsitepackagesdir}/coverage run -m mypy --ignore-missing-imports {envsitepackagesdir}/pylint/checkers/ {posargs:}
```
Comment (AWhetter) [inline tox.ini]: Please could this be run in a separate virtualenv to the tests. Seeing as mypy is failing at the moment, I think we should stop the mypy virtualenv running by default as well.

**Comment by PCManticore** [comment]
@sushobhit27 I think you might need to do a rebase since some changes that are now in master appear in your branch as well. Didn't have time yet to review this PR though

File: pylint/checkers/base.py [hunk 2]
```diff
@@ -53,12 +54,12 @@ class NamingStyle(object):
     # has multiple "accepted" forms of regular expressions,
     # but we need to special-case stuff like dunder names
     # in method names.
-    CLASS_NAME_RGX = None
-    MOD_NAME_RGX = None
-    CONST_NAME_RGX = None
-    COMP_VAR_RGX = None
-    DEFAULT_NAME_RGX = None
-    CLASS_ATTRIBUTE_RGX = None
+    CLASS_NAME_RGX = Pattern[str]
```
Comment (PCManticore) [inline pylint/checkers/base.py]: Does this class behave like `None` when we define a variable with it?

File: pylint/checkers/spelling.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
 
 try:
     import enchant
-    from enchant.tokenize import (get_tokenizer,
+    from enchant.tokenize import (get_tokenizer,    # type: ignore
```
Comment (PCManticore) [inline pylint/checkers/spelling.py]: Why do we have to ignore it here?

File: pylint/checkers/spelling.py [hunk 4]
```diff
@@ -30,11 +30,11 @@
 except ImportError:
     enchant = None
     # pylint: disable=old-style-class,no-init
-    class Filter:
+    class Filter:   # type: ignore
```
Comment (PCManticore) [inline pylint/checkers/spelling.py]: Ditto

File: pylint/checkers/utils.py [hunk 5]
```diff
@@ -104,15 +105,15 @@
 SPECIAL_METHODS_PARAMS = {
     name: params
     for params, methods in _SPECIAL_METHODS_PARAMS.items()
-    for name in methods
+    for name in methods # type: ignore
```
Comment (PCManticore) [inline pylint/checkers/utils.py]: Ditto. what does this ignore, the iteration or the result of the dictionary creation?

Comment (sushobhit27) [reply pylint/checkers/utils.py]: checkers/utils.py:108: error: "object" has no attribute "__iter__"; maybe "__str__" or "__dir__"? (not iterable)
somehow mypy not able to infer that method is a tuple. annotating _SPECIAL_METHODS_PARAMS might have fixed this, but didn't try it.

File: pylint/checkers/base.py (hunk 2 above)
Comment (sushobhit27) [reply pylint/checkers/base.py]: will check for it.

Comment (asottile) [reply pylint/checkers/base.py]: should these instead be `CLASS_NAME_RGX: Pattern[str]`

Comment (PCManticore) [reply pylint/checkers/base.py]: Unfortunately that's 3.6 syntax only so won't work for us just yet (we still support 3.4 and 3.5)

**Comment by PCManticore** [comment]
Hey @sushobhit27 Do you still plan to work on this PR? Would love to get some annotations in.

**Comment by sushobhit27** [comment]
>Hey @sushobhit27 Do you still plan to work on this PR? Would love to get some annotations in.

@PCManticore AFAIK it was decided to postpone this activity and have annotations done in astroid first, that's why I stopped working on this PR, however if you want I can continue on it. Let me know any specific files, I need to pick first.

**Comment by sushobhit27** [comment]
@PCManticore ok, I start working on it again.

**Comment by PCManticore** [comment]
We can come back later at this.

**Pull request closed without merging**